sound/soc/sof/sof-audio.h

```diff
@@ -465,6 +465,54 @@ struct snd_sof_route {
 	void *private;
 };
 
+/**
+ * struct snd_sof_dai_dma_ops - DAI DMA ops optional by default unless specified otherwise
+ * @stream_get_dma: Function pointer to get DAI DMA data
+ * @stream_put_dma: Function pointer to put DAI DMA data
+ * @setup: Function pointer for DAI DMA setup
+ * @free: Function pointer for DAI DMA free
+ * @pre_trigger: Function pointer for DAI DMA pre-trigger actions
+ * @trigger: Function pointer for DAI DMA trigger actions
+ * @post_trigger: Function pointer for DAI DMA post-trigger actions
+ */
+struct snd_sof_dai_dma_ops {
+	void *(*stream_get_dma)(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+				struct snd_pcm_substream *substream);
+	void (*stream_put_dma)(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+			       struct snd_pcm_substream *substream);
+	int (*setup)(struct snd_sof_dev *sdev, void *dma_data, unsigned int val);
+	void (*free)(struct snd_sof_dev *sdev, void *dma_data);
+	int (*pre_trigger)(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+			   struct snd_pcm_substream *substream, int cmd);
+	int (*trigger)(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+		       struct snd_pcm_substream *substream, int cmd);
+	int (*post_trigger)(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+			    struct snd_pcm_substream *substream, int cmd);
+};
+
```

Comment:
is it not for sdw dai ?

Reply:
not right now

Reply:
it's a good point though, the naming can be confusing for SoundWire. Although for SoundWire the pre and post trigger are the only things that can be configured - the trigger is handled with standard registers - and it's at the device level, not the dai level.